test/models/test_model_list_gp_regression.py

```diff
@@ -312,3 +312,69 @@ def test_fantasize(self):
             self.assertIsInstance(fm_i, SingleTaskGP)
             self.assertEqual(fm_i.train_inputs[0].shape, torch.Size([2, 8, 2]))
             self.assertEqual(fm_i.train_targets.shape, torch.Size([2, 8]))
+
+    def test_fantasize_with_outcome_transform(self) -> None:
+        """
+        Check that fantasized posteriors from a `ModelListGP` with transforms
+        relate in a predictable way to posteriors from a `ModelListGP` when the
+        outputs have been manually transformed.
+
+        We are essentially fitting "Y = 10 * X" with Y standardized.
+        - In the original space, we should predict a mean of ~5 at 0.5
+        - In the standardized space, we should predict ~0.
+        - If we untransform the result in the standardized space, we should recover
+        the prediction of ~5 we would have gotten in the original space.
+        """
+        X = torch.linspace(0, 1, 20)[:, None]
+        Y = 10 * torch.linspace(0, 1, 20)[:, None]
+        target_x = torch.tensor([[0.5]])
+
+        model_with_transform = ModelListGP(
+            SingleTaskGP(X, Y, outcome_transform=Standardize(m=1))
+        )
+        y_standardized, _ = Standardize(m=1).forward(Y)
+        model_manually_transformed = ModelListGP(SingleTaskGP(X, y_standardized))
+
+        def _get_fant_mean(model: ModelListGP) -> float:
+            fant = model.fantasize(target_x, sampler=IIDNormalSampler(10, seed=0))
+            return fant.posterior(target_x).mean.mean().item()
+
+        outcome_transform = model_with_transform.models[0].outcome_transform
+        fant_mean_with_manual_transform = _get_fant_mean(
+            model_manually_transformed
+        )  # ~0
+        manually_rescaled_mean, _ = outcome_transform.untransform(
+            fant_mean_with_manual_transform
+        )  # ~0.5
+        fant_mean_with_native_transform = _get_fant_mean(model_with_transform)  # 0.5
+
+        self.assertAlmostEqual(
+            manually_rescaled_mean.item(),
+            fant_mean_with_native_transform,
+        )
+
+    def test_fantasize_with_outcome_transform_fixed_noise(self) -> None:
+        """
+        Test that 'fantasize' on average recovers the true mean fn.
+
+        This uses a setup as close as possible to deterministic (low fixed
+        noise, fantasizing at points already seen).
+        """
+        n_fants = 20
+        y_at_low_x = 1.0
+        y_at_high_x = -0.3
+
+        X = torch.tensor([0.0, 1.0])[:, None]
+        Y = torch.tensor([y_at_low_x, y_at_high_x])[:, None]
+        yvar = torch.full_like(Y, 1e-4)
+        model = ModelListGP(
+            FixedNoiseGP(X, Y, yvar, outcome_transform=Standardize(m=1))
+        )
+
+        model.posterior(torch.zeros((1, 1)))
+
+        fant = model.fantasize(X, sampler=IIDNormalSampler(n_fants, seed=0), noise=yvar)
+
+        fant_mean = fant.posterior(X).mean.mean(0).detach().numpy().flatten()
```

Comment:
No need to make into a numpy array, can just make it into a list of python scalars. 
```suggestion
        fant_mean = fant.posterior(X).mean.mean(0).flatten().tolist()
```